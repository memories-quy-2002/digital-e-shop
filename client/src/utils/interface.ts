export enum Role {
	Customer = "Customer",
	Admin = "Admin",
}

export interface User {
	UID: number;
	UName: string;
	UEmail: string;
	UPassword: string;
	URole: Role;
	ULast_login: string;
}
