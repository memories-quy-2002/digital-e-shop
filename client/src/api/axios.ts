import axios from "axios";
declare module "axios" {
    export interface AxiosRequestConfig {
        handlerEnabled?: boolean;
    }
}

const baseURL =
    process.env.NODE_ENV === "production"
        ? "https://e-commerce-express-server-app.vercel.app/"
        : "http://localhost:4000";

export default axios.create({
    baseURL,
    withCredentials: true,
    handlerEnabled: true,
});
