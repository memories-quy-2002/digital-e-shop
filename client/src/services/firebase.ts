// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getAuth } from "firebase/auth";
// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
    apiKey: "AIzaSyCae88IRpKYJbHLxZIiArzIPYTkglQqgb0",
    authDomain: "graduation-project-5bbfb.firebaseapp.com",
    projectId: "graduation-project-5bbfb",
    storageBucket: "graduation-project-5bbfb.appspot.com",
    messagingSenderId: "503526214575",
    appId: "1:503526214575:web:5c4e1263f106bc2bee7d5a",
    measurementId: "G-NGN3CY83D3",
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
export const auth = getAuth(app);
