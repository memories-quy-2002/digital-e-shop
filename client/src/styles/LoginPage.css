.login {
  width: 90%;
  background-color: white;
  display: flex;
  flex-direction: row;
  height: 900px;
  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1), 0px 4px 8px rgba(0, 0, 0, 0.05);
}
.login:hover {
  box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.2);
}
.login__image {
  flex: 2;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  text-align: center;
}
.login__image__background {
  position: absolute;
  height: 100%;
  width: 100%;
  background: url("../assets/images/background_form.jpg");
  -o-object-fit: cover;
     object-fit: cover;
  mix-blend-mode: normal;
  filter: brightness(0.4);
}
.login__image__content {
  z-index: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}
.login__image__content__name {
  font-weight: 600;
  font-size: 60px;
  line-height: 116px;
  color: #ffffff;
  text-shadow: 0px 12px 4px rgba(0, 0, 0, 0.25);
  margin-bottom: 1rem;
}
.login__image__content__desc {
  font-weight: 500;
  font-size: 36px;
  line-height: 44px;
  color: #ffffff;
  text-shadow: 0px 12px 4px rgba(0, 0, 0, 0.25);
}
.login__form {
  height: -moz-fit-content;
  height: fit-content;
  flex: 3;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-bottom: 1rem;
}
.login__form__title {
  font-size: 36px;
  font-weight: bold;
  color: black;
  margin-top: 2rem;
  margin-bottom: 3rem;
}
.login__form__container {
  width: 80%;
}
.login__form__container__group {
  width: 100%;
}
.login__form__container__group__input {
  padding: 0.5rem;
  box-shadow: inset 0px 1px 1px 1px rgba(0, 0, 0, 0.25);
}
.login__form__container input[type=radio] {
  border: 1px solid rgba(0, 0, 0, 0.25);
}
@media (max-width: 1024px) {
  .login {
    flex-direction: column;
    height: auto;
  }
  .login__image {
    height: 300px;
  }
  .login__image__content__name {
    font-size: 50px;
    line-height: 100px;
    margin-bottom: 1rem;
  }
  .login__image__content__desc {
    font-size: 28px;
    line-height: 40px;
  }
  .login__form {
    width: 100%;
  }
  .login__form__title {
    font-size: 32px;
  }
  .login__form__container {
    width: 90%;
  }
}
@media (max-width: 768px) {
  .login {
    flex-direction: column;
    height: auto;
  }
  .login__image {
    height: 250px;
  }
  .login__image__content__name {
    font-size: 40px;
    line-height: 90px;
  }
  .login__image__content__desc {
    font-size: 24px;
    line-height: 32px;
  }
  .login__form {
    width: 100%;
  }
  .login__form__title {
    font-size: 28px;
  }
  .login__form__container {
    width: 85%;
  }
}
@media (max-width: 480px) {
  .login__image {
    height: 200px;
  }
  .login__image__content__name {
    font-size: 30px;
    line-height: 80px;
  }
  .login__image__content__desc {
    font-size: 20px;
    line-height: 28px;
  }
  .login__form {
    width: 100%;
  }
  .login__form__title {
    font-size: 24px;
  }
  .login__form__container {
    width: 80%;
  }
}/*# sourceMappingURL=LoginPage.css.map */