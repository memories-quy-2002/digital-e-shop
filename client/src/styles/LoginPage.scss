.login {
	display: flex;
	flex-direction: row;
	margin: 2rem auto;
	height: 900px;
	&__image {
		flex: 2;
		position: relative;
		&__background {
			position: absolute;
			height: 100%;
			width: 100%;
			background: url("../assets/images/background_form.jpg");
			object-fit: cover;
			mix-blend-mode: normal;
			filter: brightness(0.4);
		}
		&__content {
			width: 100%;
			height: 100%;
			&__name {
				width: inherit;
				position: absolute;
				left: 0%;
				right: 56.98%;
				top: 36.39%;
				bottom: 45.46%;
				font-weight: 600;
				font-size: 60px;
				line-height: 116px;
				text-align: center;
				color: #ffffff;
				text-shadow: 0px 12px 4px rgba(0, 0, 0, 0.25);
			}
			&__desc {
				width: inherit;
				position: absolute;
				left: 0%;
				right: 56.98%;
				top: 52.13%;
				bottom: 36.48%;
				font-weight: 500;
				font-size: 36px;
				line-height: 44px;
				text-align: center;
				color: #ffffff;

				text-shadow: 0px 12px 4px rgba(0, 0, 0, 0.25);
			}
		}
	}

	&__form {
		height: fit-content;
		flex: 3;
		display: flex;
		flex-direction: column;
		align-items: center;
		&__title {
			font-size: 36px;
			font-weight: bold;
			color: black;
			margin-bottom: 3rem;
		}
		&__container {
			width: 80%;
			&__group {
				width: 100%;
				&__input {
					padding: 0.5rem;
					box-shadow: inset 0 0 10px 2px rgba(0, 0, 0, 0.2);
				}
			}
		}
	}
}
