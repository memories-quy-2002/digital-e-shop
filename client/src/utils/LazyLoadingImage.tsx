import { useEffect, useRef } from "react";

const LazyLoadImage = ({ src, alt }: { src: string; alt: string }) => {
    const imgRef = useRef<HTMLImageElement | null>(null);

    useEffect(() => {
        const observer = new IntersectionObserver(
            (entries) => {
                entries.forEach((entry) => {
                    if (entry.isIntersecting && imgRef.current) {
                        imgRef.current.src = imgRef.current.dataset.src!;
                        observer.unobserve(imgRef.current);
                    }
                });
            },
            { threshold: 0.05 } // Load when 10% of the image is visible
        );

        if (imgRef.current) observer.observe(imgRef.current);
    }, []);

    return <img ref={imgRef} data-src={src} alt={alt} loading="lazy" />;
};
export default LazyLoadImage;
