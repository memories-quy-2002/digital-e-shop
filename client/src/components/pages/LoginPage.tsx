import { signInWithEmailAndPassword } from "firebase/auth";
import React, { useState } from "react";
import { Button, Container, Form } from "react-bootstrap";
import { useNavigate } from "react-router-dom";
import Cookies from "universal-cookie";
import axios from "../../api/axios";
import { auth } from "../../services/firebase";
import "../../styles/LoginPage.scss";
import { Role } from "../../utils/interface";
interface User {
    email: string;
    password: string;
    role: Role;
}
const LoginPage = () => {
    const navigate = useNavigate();
    const [user, setUser] = useState<User>({
        email: "",
        password: "",
        role: Role.Customer,
    });
    const [rememberMe, setRememberMe] = useState<boolean>(false);

    const [errors, setErrors] = useState<string[]>([]);

    const validateForm = (): string[] => {
        const errorsList: string[] = [];
        const emailPattern =
            /^([A-Za-z0-9_\-.])+@([A-Za-z0-9_\-.])+\.([A-Za-z]{2,4})$/;
        const passwordPattern =
            /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
        if (!user.email) {
            errorsList.push("Email is required");
        } else if (!user.email.match(emailPattern)) {
            errorsList.push("Invalid email format");
        }
        if (!user.password) {
            errorsList.push("Password is required");
        } else if (!passwordPattern.test(user.password)) {
            errorsList.push(
                "Password must be at least 8 characters long, contain at least one lowercase letter, one uppercase letter, one number, and one special character."
            );
        }
        return errorsList;
    };

    const handleChangeInput = (event: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = event.target;
        setUser({ ...user, [name]: value });
    };
    const handleChangeRadio = (event: React.ChangeEvent<HTMLInputElement>) => {
        const selectedRole = event.target.value as Role;
        setUser({ ...user, role: selectedRole });
    };
    const handleChangeCheckbox = () => {
        setRememberMe((rememberMe) => !rememberMe);
    };

    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();

        const errors = validateForm(); // Capture validation errors
        if (errors.length > 0) {
            setErrors(errors); // Display errors, no need to proceed further
            return;
        }
        try {
            const userCredential = await signInWithEmailAndPassword(
                auth,
                user.email,
                user.password
            );
            const uid = userCredential.user.uid;
            const response = await axios.post("/api/users/login", {
                uid,
                role: user.role,
            });
            if (response.status === 200) {
                const token = response.data.token;
                const cookies = new Cookies();
                const cookieData = {
                    uid,
                    token,
                };
                if (rememberMe) {
                    cookies.set("rememberMe", JSON.stringify(cookieData), {
                        httpOnly: false,
                        // Consider using Secure flag if using HTTPS
                        maxAge: 1000 * 60 * 60 * 24 * 7, // Expires in 7 days (adjust as needed)
                    });
                } else {
                    sessionStorage.setItem(
                        "rememberMe",
                        JSON.stringify(cookieData)
                    );
                }
                if (user.role === Role.Customer) {
                    navigate("/");
                } else if (user.role === Role.Admin) {
                    navigate("/admin/product");
                }
            }
        } catch (err: any) {
            console.log(err.response);

            console.error(err);
        }
    };

    return (
        <Container
            fluid
            style={{
                backgroundColor: "#E6F0FD",
                padding: "3rem",
                display: "flex",
                justifyContent: "center",
            }}
        >
            <div className="login">
                <div className="login__image">
                    <img
                        src={require("../../assets/images/background_form.jpg")}
                        alt=""
                        className="login__image__background"
                    ></img>
                    <div className="login__image__content">
                        <strong className="login__image__content__name">
                            DIGITAL-E
                        </strong>
                        <p className="login__image__content__desc">
                            An E-commerce platforms of electronics devices
                        </p>
                    </div>
                </div>
                <div className="login__form">
                    <h3 className="login__form__title">Welcome back</h3>
                    <Form
                        className="login__form__container"
                        onSubmit={handleSubmit}
                    >
                        <Form.Group
                            className="login__form__container__group mb-3"
                            controlId="formBasicUserName"
                        >
                            <Form.Label>Email</Form.Label>
                            <Form.Control
                                type="email"
                                name="email"
                                className="login__form__container__group__input"
                                required
                                autoComplete="off"
                                value={user.email}
                                onChange={handleChangeInput}
                            />
                        </Form.Group>

                        <Form.Group
                            className="login__form__container__group mb-3"
                            controlId="formBasicPassword"
                        >
                            <Form.Label>Password</Form.Label>
                            <Form.Control
                                type="password"
                                name="password"
                                className="login__form__container__group__input"
                                required
                                autoComplete="off"
                                value={user.password}
                                onChange={handleChangeInput}
                            />
                        </Form.Group>

                        <Form.Group
                            className="login__form__container__group mb-5 mt-5"
                            controlId="formBasicRadio"
                        >
                            <Form.Label>Login as:</Form.Label>
                            <br />
                            <Form.Check
                                inline
                                type="radio"
                                name="signup-role"
                                label="Customer"
                                value={Role.Customer}
                                checked={user.role === Role.Customer}
                                onChange={handleChangeRadio}
                            />
                            <Form.Check
                                inline
                                type="radio"
                                name="signup-role"
                                label="Admin"
                                value={Role.Admin}
                                checked={user.role === Role.Admin}
                                onChange={handleChangeRadio}
                            />
                        </Form.Group>
                        <Form.Group
                            className="login__form__container__group mb-3 mt-5"
                            controlId="formBasicCheckbox"
                        >
                            <Form.Check
                                inline
                                type="checkbox"
                                name="signup-role"
                                checked={rememberMe === true}
                                onChange={handleChangeCheckbox}
                                label="Remember me"
                            />
                        </Form.Group>
                        <div className="mb-5">
                            {errors.map((error, id) => (
                                <div className="text-danger mb-2" key={id}>
                                    {error}
                                </div>
                            ))}
                        </div>
                        <Button
                            variant="primary"
                            type="submit"
                            style={{
                                width: "100%",
                                height: "3rem",
                                fontSize: "16px",
                                textTransform: "uppercase",
                            }}
                        >
                            Login
                        </Button>
                        <div className="mt-4" style={{ textAlign: "center" }}>
                            Don't have an account?{" "}
                            <a href="/signup">Register</a>
                        </div>
                    </Form>
                </div>
            </div>
        </Container>
    );
};
export default LoginPage;
