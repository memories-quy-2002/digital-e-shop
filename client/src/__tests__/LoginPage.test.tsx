import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import axios from "../api/axios"; // Import your custom Axios instance
import LoginPage from "../components/pages/LoginPage"; // Adjust based on your project structure
import { Role } from "../utils/interface";
import { signInWithEmailAndPassword } from "firebase/auth";
import { auth } from "../services/firebase";
import ToastProvider from "../context/ToastContext";

// Mock Axios
jest.mock("../api/axios");
jest.mock("firebase/auth", () => ({
    getAuth: jest.fn(),
    signInWithEmailAndPassword: jest.fn(),
}));

const mockSignInWithEmailAndPassword = signInWithEmailAndPassword as jest.Mock;

// Now `axios.post` will be of type `jest.Mock`
const mockedAxios = axios as jest.Mocked<typeof axios>;
declare global {
    interface Performance {
        markResourceTiming: jest.Mock<void, []>;
    }
}

const handleOnSubmitMock = jest.fn();
describe("LoginPage", () => {
    const users = {
        customer: {
            email: "test1@gmail.com",
            password: "Phuquy2002@",
            role: Role.Customer,
        },
        admin: {
            email: "test2@gmail.com",
            password: "Phuquy2002@",
            role: Role.Admin,
        },
    };

    beforeAll(() => {
        global.performance.markResourceTiming = jest.fn();
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    const renderLoginPage = (user: typeof users.customer) => {
        render(
            <ToastProvider>
                <MemoryRouter initialEntries={["/login", "/", "/admin"]}>
                    <LoginPage />
                </MemoryRouter>
            </ToastProvider>
        );

        // Fill in email and password

        screen.getByRole("form", { name: "login-form" }).onsubmit = handleOnSubmitMock;
        fireEvent.change(screen.getByPlaceholderText(/email/i), {
            target: { value: user.email },
        });
        fireEvent.change(screen.getByPlaceholderText(/password/i), {
            target: { value: user.password },
        });

        // Select role (Customer or Admin)
        if (user.role === Role.Customer) {
            fireEvent.click(screen.getByLabelText("Customer"));
        } else if (user.role === Role.Admin) {
            fireEvent.click(screen.getByLabelText("Admin"));
        }
        fireEvent.click(screen.getByRole("button", { name: "Login" }));

        // Expectations for form submission
    };

    it("should match the LoginPage snapshot", async () => {
        const { asFragment } = render(
            <ToastProvider>
                <MemoryRouter>
                    <LoginPage />
                </MemoryRouter>
            </ToastProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("should handle successful customer login", async () => {
        // Mock axios response
        mockedAxios.post.mockImplementationOnce(() =>
            Promise.resolve({
                status: 200,
                data: { token: "customerMockedToken" },
            })
        );
        mockSignInWithEmailAndPassword.mockResolvedValueOnce({
            user: { uid: "mockedUid" }, // Mock the user UID from Firebase
        });

        renderLoginPage(users.customer);
        expect(screen.getByText("Welcome back")).toBeInTheDocument();
        expect(handleOnSubmitMock).toHaveBeenCalled();

        await waitFor(() => {
            expect(signInWithEmailAndPassword).toHaveBeenCalledWith(
                auth,
                users.customer.email,
                users.customer.password
            );
        });
        await waitFor(() => {
            expect(mockedAxios.post).toHaveBeenCalledWith("/api/users/login", {
                uid: "mockedUid",
                role: Role.Customer,
            });
        });
    });

    it("should handle successful admin login", async () => {
        // Mock axios response
        mockedAxios.post.mockImplementationOnce(() =>
            Promise.resolve({
                status: 200,
                data: { token: "adminMockedToken" },
            })
        );
        mockSignInWithEmailAndPassword.mockResolvedValueOnce({
            user: { uid: "mockedUid" }, // Mock the user UID from Firebase
        });

        renderLoginPage(users.admin);
        expect(screen.getByText("Welcome back")).toBeInTheDocument();
        expect(handleOnSubmitMock).toHaveBeenCalled();

        await waitFor(() => {
            expect(signInWithEmailAndPassword).toHaveBeenCalledWith(auth, users.admin.email, users.admin.password);
        });
        await waitFor(() => {
            expect(mockedAxios.post).toHaveBeenCalledWith("/api/users/login", {
                uid: "mockedUid",
                role: Role.Admin,
            });
        });
    });

    it("should handle login error", async () => {
        // Mock axios to return an error
        mockedAxios.post.mockImplementationOnce(() =>
            Promise.reject({
                status: 401,
                data: { token: "adminMockedToken" },
            })
        );
        renderLoginPage({
            email: "test@example.com",
            password: "wrong_password",
            role: Role.Customer,
        });
        expect(screen.getByText("Welcome back")).toBeInTheDocument();
        // Check if the error message is shown
        await waitFor(() => {
            expect(
                screen.getByText(
                    /Password must be at least 8 characters long, contain at least one lowercase letter, one uppercase letter, one number, and one special character./i
                )
            ).toBeInTheDocument();
        });
    });

    it("should handle empty email", async () => {
        // Mock axios to return an error
        mockedAxios.post.mockImplementationOnce(() =>
            Promise.reject({
                status: 401,
                data: { token: "adminMockedToken" },
            })
        );
        renderLoginPage({
            email: "",
            password: "wrong_password",
            role: Role.Customer,
        });
        expect(screen.getByText("Welcome back")).toBeInTheDocument();
        // Check if the error message is shown
        await waitFor(() => {
            expect(screen.getByText(/Email is required/i)).toBeInTheDocument();
        });
    });

    it("should handle invalid email", async () => {
        // Mock axios to return an error
        mockedAxios.post.mockImplementationOnce(() =>
            Promise.reject({
                status: 401,
                data: { token: "adminMockedToken" },
            })
        );
        renderLoginPage({
            email: "12345678",
            password: "wrong_password",
            role: Role.Customer,
        });
        expect(screen.getByText("Welcome back")).toBeInTheDocument();
        // Check if the error message is shown
        await waitFor(() => {
            expect(screen.getByText(/Invalid email format/i)).toBeInTheDocument();
        });
    });

    it("should handle empty password", async () => {
        // Mock axios to return an error
        mockedAxios.post.mockImplementationOnce(() =>
            Promise.reject({
                status: 401,
                data: { token: "adminMockedToken" },
            })
        );
        renderLoginPage({
            email: "test2@gmail.com",
            password: "",
            role: Role.Admin,
        });
        expect(screen.getByText("Welcome back")).toBeInTheDocument();
        // Check if the error message is shown
        await waitFor(() => {
            expect(screen.getByText(/Password is required/i)).toBeInTheDocument();
        });
    });
});
