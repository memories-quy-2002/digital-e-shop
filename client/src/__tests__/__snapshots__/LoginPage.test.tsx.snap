// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match the LoginPage snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
    style="background-color: rgb(230, 240, 253); padding: 3rem; display: flex; justify-content: center;"
  >
    <div
      class="login"
    >
      <aside
        class="login__image"
      >
        <img
          alt=""
          class="login__image__background"
          src="background_form.jpg"
        />
        <div
          class="login__image__content"
        >
          <strong
            class="login__image__content__name"
          >
            DIGITAL-E
          </strong>
          <p
            class="login__image__content__desc"
          >
            An E-commerce platforms of electronics devices
          </p>
        </div>
      </aside>
      <main
        class="login__form"
      >
        <h3
          class="login__form__title"
        >
          Welcome back
        </h3>
        <form
          aria-label="login-form"
          class="login__form__container"
          name="login-form"
        >
          <div
            class="login__form__container__group mb-3"
          >
            <label
              class="form-label"
              for="formBasicUserName"
            >
              Email
            </label>
            <input
              autocomplete="off"
              class="login__form__container__group__input form-control"
              id="formBasicUserName"
              name="email"
              placeholder="Email"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="login__form__container__group mb-3"
          >
            <label
              class="form-label"
              for="formBasicPassword"
            >
              Password
            </label>
            <input
              autocomplete="off"
              class="login__form__container__group__input form-control"
              id="formBasicPassword"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="login__form__container__group mb-5 mt-5"
          >
            <label
              class="form-label"
            >
              Login as:
            </label>
            <br />
            <div
              class="form-check-inline"
            >
              <input
                checked=""
                class="form-check-input"
                id="customer-radio"
                name="signup-customer"
                type="radio"
                value="Customer"
              />
            </div>
            <label
              class="me-3 form-label"
              for="customer-radio"
            >
              Customer
            </label>
            <div
              class="form-check-inline"
            >
              <input
                class="form-check-input"
                id="admin-radio"
                name="signup-admin"
                type="radio"
                value="Admin"
              />
            </div>
            <label
              class="form-label"
              for="admin-radio"
            >
              Admin
            </label>
          </div>
          <div
            class="login__form__container__group mb-3 mt-5"
          >
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                id="formBasicCheckbox"
                name="signup-role"
                type="checkbox"
              />
              <label
                class="form-check-label"
                for="formBasicCheckbox"
                title=""
              >
                Remember me
              </label>
            </div>
          </div>
          <div
            class="mb-5"
          />
          <button
            class="btn btn-primary"
            name="login"
            style="width: 100%; height: 3rem; font-size: 16px; text-transform: uppercase;"
            type="submit"
          >
            Login
          </button>
          <div
            class="mt-4"
            style="text-align: center;"
          >
            Don't have an account? 
            <a
              href="/signup"
            >
              Register
            </a>
          </div>
        </form>
      </main>
    </div>
  </div>
  <div
    class="toast-container bottom-0 end-0 p-3"
    style="z-index: 1; position: fixed; bottom: 0px; right: 0px;"
  />
</DocumentFragment>
`;
